tests/common/test_event.py

```diff
@@ -34,6 +35,36 @@
 
 
 class TestEvent(AgentTestCase):
+    def test_add_event_should_read_container_id_from_process_environment(self):
+        tmp_file = os.path.join(self.tmp_dir, "tmp_file")
+
+        def patch_save_event(json_data):
+            fileutil.write_file(tmp_file, json_data)
+
+        with patch("azurelinuxagent.common.event.EventLogger.save_event", side_effect=patch_save_event):
+            # No container id is set
+            os.environ.pop(event.CONTAINER_ID, None)
+            event.add_event(name='dummy_name')
+            data = fileutil.read_file(tmp_file)
+            self.assertIn('{"name": "ContainerId", "value": null}', data)
+
+            # Container id is set as an environment variable explicitly
+            os.environ[event.CONTAINER_ID] = '424242'
+            event.add_event(name='dummy_name')
+            data = fileutil.read_file(tmp_file)
+            self.assertIn('{{"name": "ContainerId", "value": "{0}"}}'.format(os.environ[event.CONTAINER_ID]), data)
+
+            # Container id is set as an environment variable when parsing the goal state
+            xml_text = load_data("wire/goal_state.xml")
+            goal_state = GoalState(xml_text)
+
+            container_id = goal_state.container_id
+            event.add_event(name='dummy_name')
+            data = fileutil.read_file(tmp_file)
+            self.assertIn('{{"name": "ContainerId", "value": "{0}"}}'.format(container_id), data)
+
+        os.environ.pop(event.CONTAINER_ID)
+
```

Comment:
Awesome tests! 1 suggestion - You can add 1 more case where you ensure that change in container ID in GoalState is actually modifying containerId in add_event

Reply:
Agreed, useful test case. Thanks, added!